### ISSUE TITLE
Unable to create master ssh connection

### ISSUE THREAD

**Issue by psss** [teemtee/tmt#1114]
When using `provision --how virtual` under a freshly created regular user the `provision` step fails with the following:

```
  File "/usr/lib/python3.6/site-packages/tmt/steps/provision/__init__.py", line 327, in _ssh_command
    self._ssh_master_connection(command)
  File "/usr/lib/python3.6/site-packages/tmt/steps/provision/__init__.py", line 310, in _ssh_master_connection
    command = command + self._ssh_options() + ["-MNnT", self._ssh_guest()]
  File "/usr/lib/python3.6/site-packages/tmt/steps/provision/__init__.py", line 302, in _ssh_options
    options.extend(["-S", self._ssh_socket()])
  File "/usr/lib/python3.6/site-packages/tmt/steps/provision/__init__.py", line 278, in _ssh_socket
    os.makedirs(socket_dir, exist_ok=True)
  File "/usr/lib64/python3.6/os.py", line 210, in makedirs
    makedirs(head, mode, exist_ok)
  File "/usr/lib64/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/run/user/1000'
```

The `/run/user/1000` directory is apparently not there. Any idea who takes care of making it available? Should we create the socket elsewhere?

**Comment by RHEmployee** [comment]
Hello. According to:

https://renenyffenegger.ch/notes/Linux/fhs/run/user/uid/index
Apparently, this directory is created by pam_systemd.

From man page of pam_systemd:
DESCRIPTION
       pam_systemd registers user sessions with the systemd login manager systemd-
       logind.service(8), and hence the systemd control group hierarchy.

       The module also applies various resource management and runtime parameters to the new
       session, as configured in the JSON User Records[1] of the user, when one is defined.

       On login, this module -- in conjunction with systemd-logind.service -- ensures the
       following:

        1. If it does not exist yet, the user runtime directory /run/user/$UID is either created or
           mounted as new "tmpfs" file system with quota applied, and its ownership changed to the
           user that is logging in.


Also, there is interesting (looks like realted) bug:
[Bug 967509](https://bugzilla.redhat.com/show_bug.cgi?id=967509) - The directory /run/user/ID is not created when the user is switched via su or sudo

**Comment by thrix** [comment]
Note also that there is an expectation that `tmt` runs in certain cases in a container, so I am not sure we can rely really on this directory :(

As I understand, the code now expects you are logged in a session and you run the command as the current user ...

**Comment by RHEmployee** [comment]
Then, maybe we should think about:

- detect if /run/user/uid exists
if not, then use /tmp/ instead 
if exist, just use /run/user/uid

I think we should not handle creation of /run/user/uid itself, because there is more magic (quotas, ...) involved.

**Comment by RHEmployee** [comment]
What do you think?

**Comment by psss** [comment]
Yeah, using `/run/user/uid` if it exists and `/tmp` otherwise sounds good to me.

**Comment by psss** [comment]
After adding this change we could possibly remove the `loginctl` commands from the full test?

https://github.com/teemtee/tmt/blob/a4075aec53e6c61262b6e9bb64ebb723b2db9035/tests/full/test.sh#L50

@lukaszachy, what do you say?

**Comment by RHEmployee** [comment]
First draft for review:
https://github.com/teemtee/tmt/pull/1119
